### ISSUE TITLE
Effect class constructor called only once

### ISSUE THREAD

**Issue by mpp** [mrpmorris/Fluxor#265]
Hi! I have a question regarding the Effect classes.
I see that you registered them as scoped _[here](https://github.com/mrpmorris/Fluxor/blob/65a96b579af82387ddddacb50d7c76a3b6b3b377/Source/Fluxor/DependencyInjection/ServiceRegistration/EffectClassRegistration.cs#L13)_ but I see that the constructor is called only once instead than at every dispatch of the trigger action.

For example I have two HttpClient that come from two different Nuget that I register as scoped services.
One of the two has a Dispose method that is called after every "use".

Due to this difference I found my self needing to handle them in a different way.
This for the client that do not "dispose itself":
```C#
public class FooStateEffects
{
    private readonly IState<FooStateEffects> _fooState;
    private readonly IBarHttpClient _client;

    public GlobalizationStateEffects(IBarHttpClient client, IState<FooStateEffects> fooState)
    {
        _client = client;
        _fooState= fooState;
    }

    [EffectMethod]
    public async Task HandleFooAction(FooAction action, IDispatcher dispatcher)
    {
        // Use IBarHttpClient
    }
}
```

This for the client that do "dispose itself":
```C#
public class FooStateEffects
{
    private readonly IState<FooStateEffects> _fooState;
    private readonly IServiceProvider _serviceProvider;

    public GlobalizationStateEffects(IServiceProvider serviceProvider, IState<FooStateEffects> fooState)
    {
        IServiceProvider = serviceProvider;
        _fooState= fooState;
    }

    [EffectMethod]
    public async Task HandleFooAction(FooAction action, IDispatcher dispatcher)
    {
        using var scope = _serviceProvider.CreateScope();
        var client = scope.ServiceProvider.GetService<IBazHttpClient>();
        // Use IBazHttpClient
    }
}
```

Is this the expected behavior? Am I missing/mixing something on how scoped services works?

**Comment by uhfath** [comment]
Perhaps it would be best to add some sort of an Attribute with a `ServiceLifetime` parameter so anyone could control how effects are registered.

**Comment by mpp** [comment]
> I personally would have expected effects to be registered as Transient.

I agree with you as I'd expect a new `Effect<>` instance for every dispatch of the trigger action

**Comment by mrpmorris** [comment]
Effect types are discovered at app start up and registered as scoped, the same as a store.

When a store is created via dependency injection it will create and instance of each effect type and add them to a list within the store. 

HttpClients should be reused, not disposed. If you do that, your app will quickly exhaust all available connections.

**Comment by peterbozso** [comment]
Connected discussion: #261
